pinot-core/src/main/java/com/linkedin/pinot/core/operator/MProjectionOperator.java

```diff
@@ -75,7 +81,7 @@ public ProjectionBlock getNextBlock() {
       for (String column : _columnToDataSourceMap.keySet()) {
         _blockMap.put(column, _columnToDataSourceMap.get(column).nextBlock(new BlockId(0)));
```

Comment:
we should not be calling nextBlock in the constructor. This should happen in next()